### PR TITLE
C++: Finalise docs for cpp/hresult-boolean-conversion and cpp/unsafe-dacl-security-descriptor

### PR DESCRIPTION
**Pull request by jbj**
I merged HResultBooleanConversion.qhelp and [UnsafeDaclSecurityDescriptor.qhelp](https://github.com/Semmle/ql/blob/a93939b8270bbd891949e524c93731ba3d12d5bd/cpp/ql/src/Security/CWE/CWE-732/UnsafeDaclSecurityDescriptor.qhelp) in #211 and #212, and I'd like to get docteam reviews for them in this follow-up PR.

I'll open a separate PR for change notes.

Cc @raulgarciamsft.

### CODE REVIEW

**Comment by jbj** [comment]
I've expanded the scope of this PR to include all docs work following #211 and #212. Ping @Semmle/doc.

File: change-notes/1.19/analysis-cpp.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 | **Query**                   | **Tags**  | **Purpose**                                                        |
 |-----------------------------|-----------|--------------------------------------------------------------------|
-| *@name of query (Query ID)* | *Tags*    |*Aim of the new query and whether it is enabled by default or not*  |
+| Cast between HRESULT and a Boolean type (`cpp/hresult-boolean-conversion`) | external/cwe/cwe-253 | Finds logic errors caused by mistakenly treating the Windows `HRESULT` type as a Boolean instead of testing it with the appropriate macros. Enabled by default. |
+| Setting a DACL to `NULL` in a `SECURITY_DESCRIPTOR` (`cpp/unsafe-dacl-security-descriptor`) | external/cwe/cwe-732 | This query finds code that creates world-writable objects on Windows by setting their DACL to `NULL`. Enabled by default. |
```
Comment (jbj) [inline change-notes/1.19/analysis-cpp.md]: These two queries should have the "security" tag also.

**Comment by itauthor** [comment]
LGTM

**Comment by jbj** [comment]
I just resolved the changenote conflict in the web-based conflict editor on github.com. I hope that worked.

**Comment by jbj** [comment]
@itauthor if you're happy with these changes, please approve and merge.

**Pull request merged**